impl/src/main/java/org/jboss/weld/bootstrap/enablement/EnablementListView.java

```diff
@@ -0,0 +1,163 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.weld.bootstrap.enablement;
+
+import static org.jboss.weld.util.Preconditions.checkNotNull;
+
+import java.util.List;
+import java.util.ListIterator;
+
+import javax.interceptor.Interceptor;
+
+import org.jboss.weld.exceptions.UnsupportedOperationException;
+import org.jboss.weld.util.collections.ListView;
+
+/**
+ *
+ * @author Martin Kouba
+ *
+ */
+abstract class EnablementListView extends ListView<Item, Class<?>> {
+
+    @Override
+    public boolean add(Class<?> element) {
+        checkNotNull(element);
+        List<Item> list = getDelegate();
+        synchronized (list) {
+            return list.add(createSource(element, list.isEmpty() ? null : list.get(list.size() - 1), null));
+        }
+    }
+
+    @Override
+    public Class<?> set(int index, Class<?> element) {
+        checkNotNull(element);
+        List<Item> list = getDelegate();
+        synchronized (list) {
+            if (index < 0 || index >= list.size()) {
+                throw new IndexOutOfBoundsException();
+            }
+            return toView(getDelegate().set(index, createSource(element, list.get(index).getPriority())));
+        }
+    }
+
+    @Override
+    public void add(int index, Class<?> element) {
+        checkNotNull(element);
+        List<Item> list = getDelegate();
+        synchronized (list) {
+            if (index < 0 || index >= list.size()) {
+                throw new IndexOutOfBoundsException();
+            }
+            Item previous = (index > 0) ? list.get(index - 1) : null;
+            Item next = (index <= (list.size() - 1)) ? list.get(index) : null;
+            list.add(index, createSource(element, previous, next));
+        }
+    }
+
+    @Override
+    protected Class<?> toView(Item source) {
+        return source.getJavaClass();
+    }
+
+    @Override
+    protected Item createSource(Class<?> view) {
+        throw new UnsupportedOperationException();
+    }
+
+    private Item createSource(Class<?> view, Item previous, Item next) {
+        return createSource(view, getPriority(previous, next));
+    }
+
+    private Item createSource(Class<?> view, int priority) {
+        return new Item(view, priority);
+    }
+
+    private int getPriority(Item previous, Item next) {
+        int priority;
+        if (previous == null && next == null) {
+            // No bounds
+            priority = Interceptor.Priority.APPLICATION;
```

Comment:
Right, it does not make any difference probably. It just looked weird to me to start at the boundary of the application range, instead of the middle of it.